numpy/core/tests/test_multiarray.py

```diff
@@ -478,6 +478,24 @@ def test_byteorders(self):
         assert_(np.dtype('<i4') != np.dtype('>i4'))
         assert_(np.dtype([('a', '<i4')]) != np.dtype([('a', '>i4')]))
 
+    def test_structured_non_void(self):
+        fields = [('a', '<i2'), ('b', '<i2')]
+
+        class CustomScalar(np.void):
+            pass
+        CustomScalar.__module__ = None
+
+        dt_custom = np.dtype((CustomScalar, fields))
+        str(dt_custom)  # segfault?
```

Comment:
Might as well test the output?

Reply:
Output is meaningless with `__module__` assigned to garbage, I think. Can `__module__` ever be `None` in real code?

Reply:
Yes, true. I tried it myself and indeed got `__main__.CustomScalar` - I just wondered whether this is tested anywhere at all...